mono/metadata/w32process-unix.c

```diff
@@ -2155,6 +2157,18 @@ ves_icall_System_Diagnostics_Process_ShellExecuteEx_internal (MonoW32ProcessStar
 			ret = FALSE;
 			goto done;
 		}
+
+		if (handler_needswait) {
+			gint32 exitcode;
+			MonoW32HandleWaitRet waitret;
+			waitret = process_wait (process_info->process_handle, MONO_INFINITE_WAIT, NULL);
+			ves_icall_Microsoft_Win32_NativeMethods_GetExitCodeProcess (process_info->process_handle, &exitcode);
+			if (exitcode != 0) {
+				ret = FALSE;
+				goto cleanup;
```

Comment:
Why do you need this goto?

Reply:
I don’t, code used to be different. Will fix